2.Nomic_nELISA_Analysis/1.umap/PBMC/scripts/PBMC_umap_Visualization.r

```diff
@@ -0,0 +1,112 @@
+library(ggplot2)
+library(gridExtra)
+library(cowplot)
+
+cell_type <- "PBMC"
+
+# import paths
+
+nomic_data_path <- "./results/nELISA_plate_430420_umap_PBMC.csv"
+nomic_data_selected_treatments_path <- "./results/nELISA_plate_430420_umap_PBMC_selected_treatments.csv"
+
+# read data
+nomic_data <- read.csv(nomic_data_path)
+nomic_data_selected_treatments <- read.csv(nomic_data_selected_treatments_path)
+
+head(nomic_data,2)
+head(nomic_data_selected_treatments,2)
+
+
+## plot the data
+# set the plot size
+# options(repr.plot.width=10, repr.plot.height=10)
+nomic_data_plot <- (
+    ggplot(nomic_data, aes(x = umap_1, y = umap_2, color = oneb_Metadata_Treatment_Dose_Inhibitor_Dose))
+    + geom_point(size = 5)
+    # detach the legend and plot it separately
+    + theme_bw()
+    + theme(
+        legend.position = "right",
+        axis.text = element_text(size = 10),
+        axis.title = element_text(size = 14),
+        legend.text = element_text(size = 12),
+        legend.title = element_text(size = 16),
+        strip.text = element_text(size = 8),
+        strip.background = element_rect(
+            colour = "black",
+            fill = "#fdfff4"
+            )
+            )
+    + guides(color = guide_legend(ncol = 2,title.position = 'top'))
+    + labs(title = "PBMC UMAP by treatment",color="Treatment", x = "UMAP1", y = "UMAP2")
+    # plot the legend separately
+
+
+)
+# extrtact the legend
+legend <- get_legend(nomic_data_plot)
+
+# remove the legend from the plot
+nomic_data_plot <- (
+    nomic_data_plot
+    + theme(legend.position = "none")
+)
+# show the plot + legend
+legend <- plot_grid(legend)
+nomic_data_plot
+legend
+
+# save the plot
+# set the path
+plot_path <- paste0("./figures/","umap_by_all_treatment.png")
+plot_legend_path <- paste0("./figures/","umap_by_all_treatment_legend.png")
+# save the plot
+ggsave(plot_path, nomic_data_plot, width = 8.5, height = 5.5, dpi = 500)
+ggsave(plot_legend_path, legend,dpi = 500, scale = 2)
+
+## plot the data
+# set the plot size
+# options(repr.plot.width=10, repr.plot.height=10)
+nomic_data_selected_treatments_plot <- (
+    ggplot(nomic_data_selected_treatments, aes(x = umap_1, y = umap_2, color = oneb_Metadata_Treatment_Dose_Inhibitor_Dose))
+    + geom_point(size = 5)
+    # detach the legend and plot it separately
```

Comment:
Will the detached legend be re-associated with the plot later on? Consider making a comment here in the code to help recall / indicate why this is done and how it effects the outcomes.